com.ibm.streamsx.metrics/impl/java/src/com/ibm/streamsx/metrics/internal/DeltaMetricEvaluator.java

```diff
@@ -0,0 +1,36 @@
+package com.ibm.streamsx.metrics.internal;
+
+import com.ibm.streams.management.Metric;
+
+/**
+ * This metric evaluator marks a metric as submittable if the delta between
+ * the last and the current value equals or exceeds the specified allowed
+ * delta.
+ */
+public class DeltaMetricEvaluator implements IMetricEvaluator {
+
+	private long _delta = 1;
+	
+	private Long _lastValue = null;
+	
+	public DeltaMetricEvaluator() {
+	}
+
+	public DeltaMetricEvaluator(long delta) {
+		if (delta <= 0) {
+			throw new IllegalArgumentException("DeltaMetricEvaluator(delta=" + delta + ")");
+		}
+		_delta = delta;
+	}
+	
+	@Override
+	public boolean isSubmittable(Metric metric) {
+		return (_lastValue == null) || (Math.abs(_lastValue - metric.getValue()) >= _delta);
+	}
+
+	@Override
+	public void updateStatus(Metric metric) {
+		_lastValue = Long.valueOf(metric.getValue());
```

Comment:
+1
I was not aware that Java automatically creates a new object if a long is assigned to a Long variable.